### PR TITLE
Add Alternative/MonoidK instance for AsyncStream.

### PR DESCRIPTION
**Pull request by dangerousben**

### CODE REVIEW

File: util/src/main/scala/io/catbird/util/asyncstream.scala [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
-  implicit final val asyncStreamInstances: Monad[AsyncStream] =
-    new StackSafeMonad[AsyncStream] {
+  implicit final val asyncStreamInstances: Monad[AsyncStream] with Alternative[AsyncStream] =
+    new StackSafeMonad[AsyncStream] with Alternative[AsyncStream] {
       final def pure[A](a: A): AsyncStream[A] = AsyncStream.of(a)
       final def flatMap[A, B](fa: AsyncStream[A])(f: A => AsyncStream[B]): AsyncStream[B] = fa.flatMap(f)
       override final def map[A, B](fa: AsyncStream[A])(f: A => B): AsyncStream[B] = fa.map(f)
+
+      def empty[A]: AsyncStream[A] = AsyncStream.empty[A]
+      def combineK[A](x: AsyncStream[A], y: AsyncStream[A]): AsyncStream[A] = x ++ y
```
Comment (dangerousben) [inline util/src/main/scala/io/catbird/util/asyncstream.scala:16]: Is there a rule/convention governing whether these should explicitly specify `override`, or be `final`?

Comment (felixbr) [reply util/src/main/scala/io/catbird/util/asyncstream.scala:16]: I personally like the explicit override modifier but in this project it's not used (afaik). `final` might have performance implications but I'm not an expert on it. 
In general just stick with what the existing code uses if there's no explicit styleguide.

**Comment by felixbr** [comment]
What's the status of this PR? I can't answer your questions about the priority of typeclass instances but if you don't see any immediate problems, I think we can still merge this. 🙂

**Comment by dangerousben** [comment]
> What's the status of this PR? I can't answer your questions about the priority of typeclass instances but if you don't see any immediate problems, I think we can still merge this.

It's good to go from my point of view.

**Pull request merged**